interface/main/left_nav.php

```diff
@@ -381,6 +382,71 @@ function genFindBlock()
 </table>
 <?php
 } // End function genFindBlock()
+
+/**
+ * Module URLs have an absolute URL and need to have the /interface part removed
+ * as the rest of the menu item code relies on that.
+ * @param $modUrl The full absolute url for the module
+ */
+function genModUrl($modUrl)
+{
+    return substr($modUrl, strlen('/interface') +1);
+}
+
+/**
+ * Given a module name and id, create the associated stdClass object that we can use
+ * for the menu event system.  We use these objects in other parts of OpenEMR so we
+ * conform to the same interface.
+ * @param $moduleName The name of the module
+ * @param $modId The unique id of the module.
+ * @return stdClass
+ */
+function genModuleMenuObject($moduleName, $modId)
+{
+    $moduleMenuContainer=new \stdClass();
+    $moduleMenuContainer->label=xlt($moduleName);
+    $moduleMenuContainer->url= '';
+    $moduleMenuContainer->menu_id = $modId;
+    $moduleMenuContainer->requirement=0;
+    $moduleMenuContainer->target='mod';
+    $moduleMenuContainer->children = [];
+    return $moduleMenuContainer;
+}
+
+/**
+ * Given a tree object structure of navigation items go through and generate the side navigation
+ * items for the given tree.  If the user is not an administrator than hook menu items are hidden.
+ * @param $navMenuItems  The tree object structure.  @see genModuleMenuObject for the structure of this array.
+ * @param $disallowed A hashmap of acl restrictions.
+ */
+function genModuleMenuFromMenuItems($navMenuItems, $disallowed)
+{
+    // there's only one menu item in this case
+    if (!empty($navMenuItems[0])) {
+        foreach ($navMenuItems[0]->children as $menuItem) {
+            $acl_section = $menuItem->menu_id; // we use the url as the key since it's unique...
+            if (empty($menuItem->children)) {
+                genMiscLink2('RTop', $acl_section, '0', $menuItem->label, genModUrl($menuItem->url));
+            } else if (!$disallowed['adm']) { // admin are not allowed hook settings...
+                ?>
+                <li><a class="collapsed_lv2"><span><?= $menuItem->label; ?></span></a>
```

Comment:
Not sure why, but we made decision awhile back to not use `<?=` rather than `<?php echo`
I think the logic was it was too similar to short tags

It's also better to do the html escaping where the output is done (so do not need to know the context beforehand), so would wrap this in text() and then change the xlt to just xl in the above genModuleMenuObject function.

Reply:
So the short tag expressions are always turned on in php 7, it was a problem for hosting site providers pre php 7 but I'm ok with following the community styleguide.  I'll make that change.

Also if we want the locus of translation to be at output, I'll remove the translations out of the ModuleMenuSubscriber.  That was code that I just moved from work you had done @bradymiller in src/Menu/MainMenuRole.php which did the translations inside the menu_update_map stuff.  I'll go ahead and fix that.